syntax = "proto3";

service Booking {
    rpc GetAllBookings(Void) returns (AllBookings) {}
    rpc GetBookingForUser(UserId) returns (BookingsUser) {}
    rpc AddBookingByUser(EntryAddBooking) returns (BookingsUser) {}
    rpc GetMovieAtDate(DateB) returns (ScheduleB) {}
}

message EntryAddBooking {
    UserId user_id = 1;
    NewMovie new_movie = 2;
}

message UserId {
    string id = 1;
}

message AllBookings {
    repeated BookingsUser bookings = 1;
}

message BookingsUser {
    string userid = 1;
    repeated DateItem dates = 2;
}

message DateItem {
    string date = 1;
    repeated string movies = 2;
}

message NewMovie {
    string date = 1;
    string movieid = 2;
}

message ScheduleB {
    string date = 1;
    repeated string movies = 2;
}

message DateB {
    string date = 1;
}

message Void {}